File: src/main/java/com/hileco/cortex/instructions/bits/BITWISE_AND.java
package com.hileco.cortex.instructions.bits;

import lombok.EqualsAndHashCode;

@EqualsAndHashCode(callSuper = true)
public class BITWISE_AND extends BitInstruction {
    @Override
    public byte innerExecute(byte left, byte right) {
        byte result = left;
        result &= right;
        return result;
    }
}

File: src/main/java/com/hileco/cortex/instructions/bits/BITWISE_AND.kt
package com.hileco.cortex.instructions.bits

import com.hileco.cortex.constraints.expressions.Expression
import kotlin.experimental.and

class BITWISE_AND : BitInstruction() {
    override fun innerExecute(left: Byte, right: Byte): Byte {
        return left and right
    }

    override fun innerExecute(left: Expression, right: Expression): Expression {
        return Expression.BitwiseAnd(left, right)
    }
}